src/rhash/hash_disc.c

```diff
@@ -1219,16 +1218,13 @@ static int rc_hash_wii_disc(md5_state_t* md5, const rc_hash_iterator_t* iterator
       }
     }
     else { /* Decrypted */
-      success = rc_hash_nintendo_disc_partition(md5, iterator, file_handle, (uint32_t)part_offset, 2);
+      if (rc_hash_nintendo_disc_partition(md5, iterator, file_handle, (uint32_t)part_offset, 2) == 0) {
+        return rc_hash_iterator_error(iterator, "Failed to hash Wii partition");
```

Comment:
This early return will prevent the `free(partition_table)` and `free(buffer)` calls below.